### ISSUE TITLE
Allow GitHub Actions workflow trigger for pull requests

### ISSUE THREAD

**Issue by dhs3000** [springernature/halfpipe#127]
We have some halfpipe manifests that are used to create GH Action workflows to execute tests against a pull request.

Currently, halfpipe only supports the [`push`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push) event to trigger a run. For our use case, we need to manually change the generated workflow to use [`pull_request`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) event.

It would be great if this could be somehow supported in halfpipe directly, as it makes it much easier to the developers to use only one "configuration language" (halfpipe).

**Comment by robwhitby** [comment]
yep this is an area we need to think about more, could you link to an example where you use halfpipe for PRs? I'm interested in whether halfpipe makes sense for these kind of simpler/additional workflows or maybe we should be looking more at [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) for standardisation.

**Comment by dhs3000** [comment]
There is for example these two examples:

* [workflow](https://github.com/springernature/dpas/blob/main/.github/workflows/ci-pull-request--dpas.yml) / [halfpipe file](https://github.com/springernature/dpas/blob/main/code/dpa-service/ci-pull-request.halfpipe.io.yml)
* [workflow](https://github.com/springernature/gbs-integration-order-service/blob/main/.github/workflows/ci-pull-request.yml) / [halfpipe file](https://github.com/springernature/gbs-integration-order-service/blob/main/ci-pull-request.halfpipe.io.yml)

Yes, maybe these reusable workflows could be a good fit. The workflows differ in the watched/ignored paths, injected environment variables/secrets, and the one tasks that is executed (some use docker compose, some directly a docker image&script. The latter could be moved to docker compose I assume, to reduce differences).

**Comment by efasel** [comment]
Looks like we would not be able to provide a private shared pool of reusable workflows only accessible from within the SN org. :-\ 

from [Access to reusable workflows — Reusing workflows - GitHub Docs](https://docs.github.com/en/actions/using-workflows/reusing-workflows#access-to-reusable-workflows)
>A reusable workflow can be used by another workflow if either of the following is true:
>
>* Both workflows are in the same repository.
>* The called workflow is stored in a public repository, and your organization allows you to use public reusable workflows.

On the other hand in [Overview — Reusing workflows - GitHub Docs](https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview) it says:
>Your organization can build up a library of reusable workflows that can be centrally maintained.

And in that case this might be a good fit, yes!

I guess we have to try to find out. :slightly_smiling_face:

**Comment by robwhitby** [comment]
luckily `internal` counts as `public` in this case, I already have one working:

[reusable deploy to snpaas workflow](https://github.com/springernature/ee-github-actions/blob/main/.github/workflows/snpaas-deploy.yml)

[example repo using it](https://github.com/springernature/ee-test-actions/blob/main/.github/workflows/staticfile-app.yml#L14)

I think it has good potential, we're just in a spiking phase though, there are some quirks and limitations to learn about
